addons/web/static/tests/views/fields/char_field.test.js

```diff
@@ -0,0 +1,847 @@
+        message: "input should contain correct value after onchange"
+    });
+});
+
+test("char field with placeholder", async () => {
+    Partner._fields.foo = fields.Char({ string: "Foo", default: false });
+    await mountView({ type: "form", resModel: "res.partner", arch: `
+        <form>
+            <sheet>
+                <group>
+                    <field name="foo" placeholder="Placeholder" />
+                </group>
+            </sheet>
+        </form>`
+    });
+    expect(".o_field_widget[name='foo'] input").toHaveAttribute("placeholder", "Placeholder", {
+        message: "placeholder attribute should be set"
+    });
+});
+
+test("char field: correct value is used to evaluate the modifiers", async () => {
+    Partner._records[0].foo = false;
+    Partner._records[0].display_name = false;
+    Partner._fields.foo = fields.Char({ string: "Foo", onChange: (obj) => {
+        if (obj.foo === "a") {
+            obj.display_name = false;
+        } else if (obj.foo === "b") {
+            obj.display_name = "";
+        }
+    }});
+    await mountView({ type: "form", resModel: "res.partner", resId: 1, arch: `
+        <form>
+            <field name="foo" />
+            <field name="display_name" invisible="'' == display_name"/>
+        </form>`,
+    });
+    expect("[name='display_name']").toHaveCount(1);
+
+    await fieldInput("foo").edit("a");
+    await animationFrame();
+    expect("[name='display_name']").toHaveCount(1);
+
+    await fieldInput("foo").edit("b");
+    await animationFrame();
+    expect("[name='display_name']").toHaveCount(0);
+});
+
+test("edit a char field should display the status indicator buttons without flickering", async () => {
+    Partner._records[0].p = [2];
+    Partner._fields.foo = fields.Char({ string: "Foo", onChange: (obj) => {obj.display_name = "cc";}});
+
+    const def = new Deferred();
+    await mountView({ type: "form", resModel: "res.partner", resId: 1, arch: `
+        <form>
+            <field name="p">
+                <tree editable="bottom">
+                    <field name="foo"/>
+                </tree>
+            </field>
+        </form>`
+    });
+    onRpc(async (route, { method, args, kwargs}) => {
+        debugger;
+        if (method === "onchange") {
+            expect.step("onchange");
+            await def;
+        }
+    });
+    expect(".o_form_status_indicator_buttons.invisible").toHaveCount(1, {
+        message: "form view is not dirty"
+    });
+    await contains(".o_data_cell").click();
+    await fieldInput("foo").edit("a");
+    expect(".o_form_status_indicator_buttons:not(.invisible)").toHaveCount(1, {
+        message: "form view is dirty"
+    });
+    def.resolve();
+    expect(["onchange"]).toVerifySteps();
+    await animationFrame();
+    expect(".o_form_status_indicator_buttons:not(.invisible)").toHaveCount(1, {
+        message: "form view is dirty"
+    });
+    expect(["onchange"]).toVerifySteps();
```

Comment:
This is a behaviour modification of the test. There is now two "onchange" step in this test.

Before
-------
When calling `editInput`, the mechanism is that the input is modified and then the `input` and `change` events are triggered.
This cause one onchange.

Now
----
Now the new helper `await fieldInput("foo").edit("a");` modify the input and press `Enter`.
This cause two onchange:
1) the onchange of the modified line
2)  the empty onchange of the newline cause `edit` press `Enter`